### PR TITLE
Add Rooms app 

### PR DESCRIPTION
**Pull request by bartacc**
This PR contains reimplementation of the Rooms frontend app. 

Old version was also written in React (check it [here](https://github.com/ksiuwr/zosia-site-v2016/blob/master/app/js/room.js)), but without Typescript and React Query.

### CODE REVIEW

File: app/client/utils/roomData.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { templates } from "@reactivated";
+import { parseISO } from "date-fns/parseISO";
+
+export const ROOM_QUERY_KEY = "rooms";
+
+/** Single room data retrieved from server */
+export interface RoomAPIData {
```
Comment (ref-humbold) [inline app/client/utils/roomData.ts:7]: Please decide whether you use `API` (like here) or `Api` (like in React component names) - I personally recommend the latter.

File: app/client/components/LoadingContentSpinner.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+interface LoadingContentSpinnerProps {
+  isLoading: boolean;
+}
+
+export const LoadingContentSpinner = ({
+  isLoading,
+  children,
+}: PropsWithChildren<LoadingContentSpinnerProps>) => {
+  if (isLoading) {
```
Comment (ref-humbold) [inline app/client/components/LoadingContentSpinner.tsx:11]: You could simply use ternary operator: `return isLoading ? (<span ...>) : (<>{children}</>)`

File: app/client/utils/roomData.ts (hunk 1 above)
Comment (bartacc) [reply app/client/utils/roomData.ts]: Done, I changed all of them to `Api`.

File: app/client/components/LoadingContentSpinner.tsx (hunk 2 above)
Comment (bartacc) [reply app/client/components/LoadingContentSpinner.tsx]: Done

**Pull request merged**